app/javascript/components/i18n/provider.unit.test.js

```diff
@@ -0,0 +1,41 @@
+import React from "react";
+import { Provider } from "react-redux";
+
+import { createSimpleMount } from "../../test";
+import store from "../../store";
+
+import I18nProvider from "./provider";
+
+describe("I18nProvider - changeLocale", () => {
+  it("should have defaults for window and document", () => {
+    const defaultLocale = "en";
+
+    const { locale } = window.I18n;
+    const { lang } = document.documentElement;
+
+    expect(locale).to.equal(defaultLocale);
+    expect(lang).to.equal(defaultLocale);
+  });
+
+  it("should handle changeLocale method", () => {
+    const newLocale = "fr";
+
+    const component = createSimpleMount(
+      <Provider store={store()}>
+        <I18nProvider>
+          <div>wrapped children</div>
+        </I18nProvider>
+      </Provider>
+    );
+
+    console.log(component.find(I18nProvider).first().instance()); // null
+
+    component.find(I18nProvider).first().instance().changeLocale(newLocale);
```

Comment:
Is component.find(I18nProvider).first() returning the correct component?

Reply:
No, it's not, it's `null`, I tried different variations and they're all `null` and I'm not sure how I should be mounting it.